cluster/examples/kubernetes/ceph/cluster.yaml

```diff
@@ -237,8 +237,10 @@ spec:
       journalSizeMB: "1024"  # this value can be removed for environments with normal sized disks (20 GB or larger)
       osdsPerDevice: "1" # this value can be overridden at the node or device level
 # Cluster level list of directories to use for storage. These values will be set for all nodes that have no `directories` set.
-#    directories:
-#    - path: /rook/storage-dir
+    directories:
+    # By default create a osd in the dataDirHostPath directory. This should be removed for
+    # environments where nodes have disks available for Rook to use.
+    - path: /var/lib/rook
```

Comment:
have you been able to test this on a variety of environments?  i'm wondering what the first experience will be for folks that don't make any edits to this yaml file and just try it as is.

Reply:
The environment I have is a 3-node-1-master kubeadm cluster running k8s v1.12.4. What kind of other environments do you have in mind? 

Although you have me thinking about stuff now, and I have something I want to try with the yaml file.

Reply:
No dice. I tried making a yaml reference variable to automatically use whateverthe value of `dataDirHostPath` is, but it didn't get picked up.
```yaml
# ...
dataDirHostPath: &dataDirHostPath /var/lib/rook
# ...
storage:
  directories:
  - path: *dataDirHostPath
# ...
```

Reply:
I was mostly thinking of the environments that have issues with writeable filesystems, e.g. the ones listed in https://rook.io/docs/rook/v0.9/flexvolume.html. were you able to try 1 of those?

Reply:
Ahh. I see. It seems like there isn't currently mention of the `/var/lib/rook` dir not working on any particular environments. I have no doubt that there are environments where `/var/lib/rook` isn't writeable, including SUSE's CaaSP 3.0 product :stuck_out_tongue_winking_eye:. For the new double-instance of `/var/lib/rook`, I'm not sure how best to tell the user those two fields are tied together for the example, but they can be separate if the user likes.

It might be better if no devices and no dirs are specified in the example CRD at all and have the quickstart include the requirement of specifying at least one of these. I don't know. 

I'm not sure if there is a path in all of those environments that is writable ... maybe `/home/rook`? In CaaSP /home is an overlayfs, but I don't know about the others. Does anyone know? I feel like this could be an issue for a small PR.

Reply:
I think this can be addressed by separating example production yamls from the simple single-node test yamls. Addressing it independently from this PR seems prudent. @jbw976 sound ok?